### PR TITLE
clean up

### PR DESCRIPTION
**Pull request by javasoze**
@leleuj 

This PR contains the following:

  * fix a potential NPE when clientsProperties are not initialized
  * Added heathcheck boilerplate to void seeing the annoying warning
  * In pom.xml using the Shade plugin to build a fat jar (the recommended Dropwizard way), so you can specify config .yml files

To build and run:

Build:

    mvn clean package

this will build a fat jar into target/

Run:

    java -jar target/dropwizard-pac4j-demo-1.0.0-SNAPSHOT.jar server demo-config2.yml <- can be any yml

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -97,6 +81,43 @@
           <target>${java.version}</target>
         </configuration>
       </plugin>
+      <!-- Shade -->
```
Comment (leleuj) [inline pom.xml]: Why did you remove the `exec-maven-plugin` Maven plugin? It seems a convenient way to launch the demo using: `mvn compile exec:exec`. How do we do now?

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java [hunk 2]
```diff
@@ -48,6 +50,12 @@ protected void configure() {
         });
 
         env.jersey().register(ViewsResource.class);
-    }
 
+        env.healthChecks().register(getName(), new HealthCheck() {
```
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java]: A health check is always a good practice for production, I don't think it's really useful for a demo where the primary aim is to display the UI and click on links to understand things, but anyway, we can have it.

File: src/main/java/org/pac4j/demo/dw/factory/Pac4jFactory2.java [hunk 3]
```diff
@@ -6,22 +6,26 @@
...
+    private Map<String, String> clientsProperties = new HashMap<>();
 
     public Map<String, String> getClientsProperties() {
         return clientsProperties;
     }
 
     @JsonProperty
     public void setClientsProperties(Map<String, String> clientsProperties) {
-        this.clientsProperties = clientsProperties;
+	    if (clientsProperties != null) {
```
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/factory/Pac4jFactory2.java]: Yes, it's better that way, it makes the `clientsProperties` field optional. This class is just a draft for tests, I will refactor it properly before submitting a PR to the dropwizard-pac4j lib.

File: pom.xml (hunk 1 above)
Comment (javasoze) [inline pom.xml]: @leleuj see my PR description:

build

    mvn clean package

a fat jar will be created in the target dir

To run:

    java -jar target/dropwizard-pac4j-demo-1.0.0-SNAPSHOT.jar server demo-config2.yml

The reason I chose to use the shade plugin to build a fat jar is so that we can run the demo by setting different configs without rebuilding/re-editing the pom

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java (hunk 2 above)
Comment (javasoze) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java]: yeah, this is unnecc. I was just bothered by the warning :)

File: pom.xml (hunk 1 above)
Comment (leleuj) [inline pom.xml]: OK. I see your point. I don't really think we want to switch from YAML file very often as it's just here to demonstrate two different ways for configuration. I think it's faster to run `mvn clean compile exec:exec` than producing a JAR. I would stick to the current way, but I don't have a strong opinion on this. Maybe we can have both.

File: src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java (hunk 2 above)
Comment (leleuj) [inline src/main/java/org/pac4j/demo/dw/Pac4JDemoApplication.java]: OK. I see it now. You're right.

File: pom.xml (hunk 1 above)
Comment (leleuj) [inline pom.xml]: @victornoel: what do you think?

Comment (victornoel) [inline pom.xml]: I would keep the exec plugin nevertheless, I usually find it useful to run stuff quickly without running many commands and tests and packaging.

Also, I personally prefer to use https://github.com/chrisdchristo/capsule-maven-plugin to build autoexecutable jars, it's cleaner and nicer, but it's just my personal preference.

**Comment by victornoel** [comment]
apart for the `clientProperties` problem, this could be merged in master (once exec is put back if desired and a shade is confirmed as the way to make fat jar).

File: pom.xml (hunk 1 above)
Comment (javasoze) [inline pom.xml]: sounds good. I restored the plugin section. Sorry for causing a problem.

Comment (victornoel) [inline pom.xml]: no problem, thanks for proposing improvements :)

**Comment by javasoze** [comment]
@leleuj I have restored the exec plugin in the pom. Apologies again for removing it. Are you ok with merging this into your branch?

**Comment by leleuj** [comment]
Everything is ok for me. Merging.
